student-store-express-api/package.json

```diff
@@ -4,7 +4,7 @@
   "main": "server.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "node server.js",
+    "start": "nodemon server.js",
```

Comment:
how does nodemon prevent an infinite loop?

Reply:
the package uses a couple of mechanisms to prevent an infinite loop including ; delay before restarting, debouncing files changes and also monitoring specific file extensions

Reply:
but what was the cause for the infinite loops in the first place?